debian/control

```diff
@@ -300,6 +300,7 @@ Depends: ${misc:Depends},
          ${shlibs:Depends},
          python3-dateutil,
          python3-openssl,
+         python3-typing-extensions
```

Comment:
this package does not exist in ubuntu/bionic on which we perform tests using rados suites. see https://packages.ubuntu.com/search?keywords=python3-typing-extensions&searchon=names&suite=bionic&section=all . it exists in later releases of this distro though, see https://packages.ubuntu.com/search?keywords=python3-typing-extensions&searchon=names&suite=all&section=all

is it possible to you to make this dependency optional ?

Reply:
Sure, @tchaikov, but then could we run [Python 3.7/3.8 in those releases](http://ubuntuhandbook.org/index.php/2020/07/python-3-8-4-released-install-ubuntu-18-04-16-04/)? Because 3.6 lacks lots of `typing` features, and hence the `extensions` backport.

Alternatively, do you know about other ways of fixing this? In the end, `typing-extensions` is just [a single Python-only file](https://github.com/python/typing/blob/master/typing_extensions/src_py3/typing_extensions.py)...

Reply:
> Sure, @tchaikov, but then could we run [Python 3.7/3.8 in those releases](http://ubuntuhandbook.org/index.php/2020/07/python-3-8-4-released-install-ubuntu-18-04-16-04/)? Because 3.6 lacks lots of `typing` features, and hence the `extensions` backport.

apart from setting up our testing infra for this new dependency, that'd be a breaking change which forces user to install python3.7/python3.8 from a 3rd party PPA if they are using ubuntu bionic.

> 
> Alternatively, do you know about other ways of fixing this? In the end, `typing-extensions` is just [a single Python-only file](https://github.com/python/typing/blob/master/typing_extensions/src_py3/typing_extensions.py)...

sorry, i don't have enough bandwidth looking into this issue. but since you answered "Sure". did you mean that you could make this optional?

Reply:
> sorry, i don't have enough bandwidth looking into this issue. but since you answered "Sure". did you mean that you could make this optional?

@tchaikov sorry for the misunderstanding, I rather meant that I'd check whether it's possible or not to make this optional... Unfortunately, as since Python 3 type hints are no longer comment annotations (as in Python 2.7), they're evaluated by the Python interpreter as well (and not only `mypy` static checker) and hence require that all definitions there (like `Literal`) are implemented as code. I'll check how this can be worked around...

As it seems that OS release support is (at least for me) a topic where I often stumble upon, so is it published anywhere the distro-release support matrix for Ceph?

Thanks!

Reply:
two ideas:

```python
from typing import TYPE_CHECKIG
if TYPE_CHECKING  # always False at runtime
    from typing_extensions import Literal

always_true: "Literal[True]" = True

# or

always_true = True  # type: Literal[True]

Reply:
> 
> As it seems that OS release support is (at least for me) a topic where I often stumble upon, so is it published anywhere the distro-release support matrix for Ceph?
> 

see https://ceph.readthedocs.io/en/latest/start/os-recommendations/?highlight=OS%20Recommendations. JFYI, i also found that RHEL8/CentOS8 comes with Python 3.6.8.

Reply:
@sebastian-philipp thanks a lot! Just followed that approach.

@tchaikov thanks a lot for pointing me to that!